apps/aecore/src/aec_chain_state.erl

```diff
@@ -112,6 +113,13 @@ top_block(?match_state(top_block_hash := undefined)) -> undefined;
 top_block(?match_state(top_block_hash := X) = State) ->
     export_block(blocks_db_get(X, State), State).
 
+-spec top_of_chain(non_neg_integer(), state()) ->
+                          {'ok', list(#header{})} | {error, atom()}.
+top_of_chain(_N, ?match_state(top_header_hash := undefined)) ->
```

Comment:
I would really prefer a less ambiguous name for this function. get_n_headers_from_top or something.